logic/pulsed/sequence_generator_logic.py

```diff
@@ -1896,6 +1913,11 @@ def sample_pulse_sequence(self, sequence):
 
                 # Add created waveform names to the set
                 written_waveforms.update(waveform_list)
+            else:
```

Comment:
This `else:` clause must do much more than just taking the information from the ensemble instance as granted.
The case where the ensemble has been sampled with different pulse generator settings has to be handled. So each parameter in the `sampling_information` dict has to be sanity checked if it is still valid (active channels etc.).

Reply:
did a check, that the pulse_generator_settings are really the same.

Reply:
Looks ok... I'm no big fan of those unreadable logic expressions though.